github_discovery.py

```diff
@@ -187,6 +189,31 @@ def is_ipallowList_enabled(yaml_data):
         ip_allow_list_enabled = True
   return ip_allow_list_enabled
 
+def get_trivy_scan_json_data(project_name):     
+  circleci_headers = {"Authorization": f"Circle-Token: {CIRCLECI_TOKEN}", "Content-Type": "application/json", "Accept": "application/json"}
+  project_url = f"{CIRCLECI_API_ENDPOINT}"+ project_name
+  output_json_content={}
+  try:
+    response = requests.get(project_url, headers=circleci_headers) 
+    for build_info in response.json():
+      if build_info.get('workflows',{}).get('workflow_name') == 'security':
+        latest_build_num = build_info['build_num'] 
+        artifacts_url = f"{project_url}/{latest_build_num}/artifacts" 
+        break
+    response = requests.get(artifacts_url, headers=circleci_headers) 
+    artifact_urls = response.json()                     
+    output_json_url = next((artifact['url'] for artifact in artifact_urls if 'results.json' in artifact['url']), None)
+    if output_json_url:
+      response = requests.get(output_json_url, headers=circleci_headers) 
+      output_json_content = response.json()
+      return output_json_content
```

Comment:
Also doing this in the `else`, you could scrap the else and move this line after the `if` ?

Reply:
This if ensures that in case there are multiple files in Artifact directory , then it gets executed only if there exist a results.json file. Else, it returns None.